fast64_internal/f3d/f3d_writer.py

```diff
@@ -334,14 +334,17 @@ def saveStaticModel(
 
     # checkForF3DMaterial(obj)
 
-    facesByMat = {}
+    faces_by_mat = {}
     for face in obj.data.loop_triangles:
-        if face.material_index not in facesByMat:
-            facesByMat[face.material_index] = []
-        facesByMat[face.material_index].append(face)
+        if face.material_index not in faces_by_mat:
+            faces_by_mat[face.material_index] = []
+        faces_by_mat[face.material_index].append(face)
+
+    # sort by material slot
+    faces_by_mat = {mat_index: faces_by_mat[mat_index] for mat_index, _ in enumerate(obj.material_slots)}
```

Comment:
```suggestion
    faces_by_mat = {mat_index: faces_by_mat[mat_index] for mat_index, _ in enumerate(obj.material_slots) if mat_index in faces_by_mat}
```
this isn't just theoretical, this errors if you have an empty material slot e.g. :
![image](https://github.com/Fast-64/fast64/assets/5306619/c658e827-b9c6-4a5a-a36a-53315a6b7dcd)

or even a material that is assigned to no face